R/extractCol.R

```diff
@@ -66,7 +66,11 @@ extractCol <- function(contrastList, colName, robust = TRUE){
 
     for (i in 1:length(contrastList)) {
         newdat <- cbind(rowid = rownames(contrastList[[i]]), data.frame(contrastList[[i]], row.names = NULL))
-        newdat <- newdat[, c("rowid", "P.Value")]
+        # use the pass colName to extract related data
+        newdat <- newdat[, c("rowid", colName)]
+        # rename the colname to match the list name to avoid duplication after
+        # merging the same column over and over
```

Comment:
It merges the same column from different data.frame (list of data.frame) so I don't think you need to name them in loop and it's already renamed at line 82. please compare output (I don't think you need this code )

Reply:
@kar-agg-gen - because it merges the same column each time (and here we are merging columns not rows) the column will have two **columnName.x** and two **columnName.y** hence the warning and other issues.
Please see https://3.basecamp.com/3775257/buckets/16965597/todos/3419789715 comments. Thank you

Reply:
I get it

Reply:
You can remove line 82 then its already done in loop

Reply:
done